tenpy/algorithms/truncation.py

```diff
@@ -307,6 +307,330 @@ def svd_theta(theta, trunc_par, qtotal_LR=[None, None], inner_labels=['vR', 'vL'
+        If `True` and ``move_right=True``, the right tensor `T_Rc` is returned in `'Th'` (`'B'`) form, 
+        if ``use_eig_based_svd=True`` (``use_eig_based_svd=False``).
+        If `True` and ``move_right=False``, the left tensor `T_Lc` is returned in `'Th'` (`'A'`) form, 
+        if ``use_eig_based_svd=True`` (``use_eig_based_svd=False``).
+
+    Returns
+    -------
+    T_Lc : array with legs [(vL.p), vR] or None
+    S : 1D numpy array
+        The singular values of the array.
+        Normalized to ``np.linalg.norm(S)==1``.
+    T_Rc : array with legs [vL, (p.vR)] or None
+    form : list
+        List containing two entries providing the form of the two arrays `T_Lc` and `T_Rc` in string form.
+        e.g. ``['A','Th']``
+    trunc_err : TruncationError
+    renormalization : float
+        Factor, by which S was renormalized.
+    """
+
+    if compute_err:
+        return_both_T = True
+    
+    if move_right:
+        # Get initial guess for the left isometry
+        Y0 = _qr_theta_Y0(old_qtotal_L, old_qtotal_R, old_bond_leg, theta, move_right, expand, min_block_increase) # Y0: [(vL.p0), vR]
+
+        # QR based updates
+        theta_i1 = npc.tensordot(Y0.conj(), theta, ['(vL*.p0*)', '(vL.p0)']).ireplace_label('vR*', 'vL') # theta_i1: [vL,(p1.vR)]
+        theta_i1.itranspose(['(p1.vR)', 'vL']) # theta_i1: [(p1.vR),vL]
+        B_R, _ = npc.qr(theta_i1, inner_labels=['vL', 'vR'], inner_qconj=-1) # B_R: [(p1.vR),vL] 
+        B_R.itranspose(['vL', '(p1.vR)']) # B_R: [vL,(p1.vR)] 
+
+        theta_i0 = npc.tensordot(theta, B_R.conj(), ['(p1.vR)', '(p1*.vR*)']).ireplace_label('vL*', 'vR') # theta_i0: [(vL.p0),vR]
+        A_L, Xi = npc.qr(theta_i0, inner_labels=['vR', 'vL']) # A_L: [(vL.p0), vR]
+        
+    else:
+        # Get initial guess for the right isometry
+        Y0 = _qr_theta_Y0(old_qtotal_L, old_qtotal_R, old_bond_leg, theta, move_right, expand, min_block_increase) # Y0: [vL, (p1.vR)]
+
+        # QR based updates
+        theta_i0 = npc.tensordot(theta, Y0.conj(), ['(p1.vR)', '(p1*.vR*)']).ireplace_label('vL*', 'vR') # theta_i0: [(vL.p0),vR]
+        A_L, _ = npc.qr(theta_i0, inner_labels=['vR', 'vL']) # A_L: [(vL.p0), vR]
+
+        theta_i1 = npc.tensordot(A_L.conj(), theta, ['(vL*.p0*)', '(vL.p0)']).ireplace_label('vR*', 'vL') # theta_i1: [vL,(p1.vR)]
+        theta_i1.itranspose(['(p1.vR)', 'vL']) # theta_i1: [(p1.vR),vL]
+        B_R, Xi = npc.qr(theta_i1, inner_labels=['vL', 'vR'], inner_qconj=-1)
+        B_R.itranspose(['vL', '(p1.vR)'])
+        Xi.itranspose(['vL', 'vR'])
+
+    # SVD of bond matrix Xi
+    if use_eig_based_svd:
+        U, S, Vd, _, renormalization = _eig_based_svd(
+            Xi, need_U=move_right, need_Vd=(not move_right), inner_labels=['vR', 'vL'], trunc_params=trunc_params
+        )
+    else:
+        U, S, Vd, _, renormalization = svd_theta(Xi, trunc_params) # <- TODO: Is it fine to not specify the charges here?
```

Comment:
Per default, this means `qtotal_R = Xi.qtotal` (see [here](https://github.com/tenpy/tenpy/blob/224e6cf1e7b00480e4d51047d5b0a6beb7959fa3/tenpy/linalg/np_conserved.py#L3576)). As long as `Xi.qtotal=0` it does not matter, right? Could there be a case where `Xi.qtotal!=0` and we suddenly need to care about `qtotal_LR`?